tests/bgp/test_passive_peering.py

```diff
@@ -0,0 +1,198 @@
+'''
+
+This script is to test BGP passive peering on SONiC.
+
+'''
+
+import logging
+import pytest
+import time
+from tests.common.config_reload import config_reload
+from tests.common.helpers.constants import DEFAULT_NAMESPACE
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+bgp_config_sleeptime = 90
+peer_password = "sonic.123"
+wrong_password = "wrong-password"
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, request):
+    # verify neighbors are type sonic
+    if request.config.getoption("neighbor_type") != "sonic":
```

Comment:
another Q is why this test requires neighbor to be sonic?

Reply:
I do this check because configuration changes to the neighbor need to use the correct syntax and I need to use vsonic neighbors for my testbed.  Which means I have only tested the script with sonic neighbors.

Reply:
we don't have vsonic neighbor in lab, so this test and the bgp authentication test will be noop for our testing.

could you make it comprehensive for eos/sonic neighbor? e.g. for the `is_multi_asic` check you can skip for eos, deal with them as single-asic. for any function you want to use for eos, please refer to tests/common/devices/eos.py

Reply:
I can try, but I don't have a way to test EOS neighbors and their commands so it would just be guess work.

Reply:
@wenyiz2021 I updated as I could, but it is untested.

Reply:
thanks @Azarack let me verify